transport/src/test/java/io/netty/channel/ManualIoEventLoopTest.java

```diff
@@ -216,6 +218,33 @@ public Boolean call() {
         }
     }
 
+    @Test
+    public void testDelayOwningThread() throws ExecutionException, InterruptedException {
+        Semaphore semaphore = new Semaphore(0);
+        ManualIoEventLoop eventLoop = new ManualIoEventLoop(null, executor ->
+                new TestIoHandler(semaphore));
+        Thread thread = new Thread(() -> {
+            eventLoop.setOwningThread(Thread.currentThread());
+            assertTrue(eventLoop.inEventLoop());
+            while (!eventLoop.isTerminated()) {
+                eventLoop.runNow();
+            }
+        });
+
+        assertFalse(eventLoop.inEventLoop());
+
+        CompletableFuture<Void> cf = new CompletableFuture<>();
```

Comment:
verify that `eventLoop::run` is throwing an exception too (you can add a separate test for it)

Reply:
You mean verify the `owningThread == null` check?

Reply:
The check you added in run to make it fail - yep